### PR TITLE
[GSoC 2024] Integrating KinFin Proteome Cluster analyses into Genome Browsing environments

### PR DESCRIPTION
**Pull request by rohan-b-84**
The below issues are part of GSoC 2024 work product submissions.
- [x] #1 
- [x] #2 
- [x] #4 
- [x] #5 
- [x] #6 
- [x] #7 
- [x] #8 
- [x] #16 
- [x] #17 

The following issue(s) would be worked upon even after completion of GSoC 2024. 
- [ ] #18

### CODE REVIEW

File: requirements.txt
Comment (rjchallis) [inline requirements.txt:1]: would be nice to split dependencies and move pytest into `requirements-dev.txt`. Can have `-r requirements.txt` as the first line of this file so `pip install -r requirements-dev.txt` will install everything

Comment (rohan-b-84) [reply requirements.txt:1]: done

File: requirements-dev.txt
Comment (rjchallis) [inline requirements-dev.txt:1]: first line can be 
```
-r requirements.txt
```
then only pytest dep is needed to save maintaining versions in 2 files

File: src/cli/validate.py
Comment (rjchallis) [inline src/cli/validate.py:1]: Ideally use a logging module for these messages

File: src/api/__init__.py
Comment (rjchallis) [inline src/api/__init__.py:1]: add a docstring to describe the function

File: src/api/endpoints.py
Comment (rjchallis) [inline src/api/endpoints.py:1]: add a docstring to describe the function

File: src/api/sessions.py
Comment (rjchallis) [inline src/api/sessions.py:1]: add docstrings

File: src/cli/__init__.py
Comment (rjchallis) [inline src/cli/__init__.py:1]: add a docstring to describe the function

File: src/core/build.py
Comment (rjchallis) [inline src/core/build.py:1]: would prefer logging to print

File: src/core/datastore.py
Comment (rjchallis) [inline src/core/datastore.py:1]: Would be better to replace long series of `append`s with list declaration

File: src/core/results.py
Comment (rjchallis) [inline src/core/results.py:1]: add a docstring to describe the function

File: src/core/utils.py
Comment (rjchallis) [inline src/core/utils.py:1]: add missing docstrings

File: src/main.py
Comment (rjchallis) [inline src/main.py:1]: use logging instead of print

File: test
Comment (rjchallis) [inline test:1]: would prefer `.sh` suffix on shell scripts and consider moving test scripts/files to tests subdirectory

**Review by rjchallis — CHANGES_REQUESTED** [review]
This is looking good! I don't know the code well enough to comment on details, so mostly comments about docstrings and ideally using logging instead of print statements.

**Review by rjchallis — APPROVED** [review]

**Comment by rjchallis** [comment]
Looks great - lets get this merged and save the full API docs for a follow up PR

**Pull request merged**